File: code/code/cpp/78_subsets.cpp
#include <vector>

class Solution {
public:
    std::vector<std::vector<int>> subsets(std::vector<int>& nums) {
        std::vector<std::vector<int>> result;
        std::vector<int> current;
        backtrack(nums, 0, current, result);
        return result;
    }

private:
    void backtrack(const std::vector<int>& nums,
                   int start,
                   std::vector<int>& current,
                   std::vector<std::vector<int>>& result) {
        result.push_back(current);
        for (int i = start; i < nums.size(); ++i) {
            current.push_back(nums[i]);
            backtrack(nums, i + 1, current, result);
            current.pop_back();
        }
    }
};

File: code/1. 纯文本的文件保存位置  
./code/rust/78_subsets.rs
```rust
impl Solution {
    pub fn subsets(nums: Vec<i32>) -> Vec<Vec<i32>> {
        let mut res = Vec::new();
        let mut cur = Vec::new();
        fn backtrack(start: usize, nums: &[i32], cur: &mut Vec<i32>, res: &mut Vec<Vec<i32>>) {
            res.push(cur.clone());
            for i in start..nums.len() {
                cur.push(nums[i]);
                backtrack(i + 1, nums, cur, res);
                cur.pop();
            }
        }
        backtrack(0, &nums, &mut cur, &mut res);
        res
    }
}
```